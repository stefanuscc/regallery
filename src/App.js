/* Libraries */
import { useState } from 'react'
import Cookies from 'universal-cookie';
import './App.scss'

/* Hooks */
import { useFetch } from './hooks/useFetch'

/* Components */
import Alert from './components/micro/Alert/Alert'
import List from './components/List/List'
import Gallery from './components/Gallery/Gallery'
import ModeControl from './components/Control/ModeControl/ModeControl'
import PageControl from './components/Control/PageControl/PageControl'
import Modal from './components/micro/Modal/Modal'
import Input from './components/micro/Input/Input'
import ModalControl from './components/Control/ModalControl/ModalControl'

function App() {
  const endPoint = "https://picsum.photos/v2/list"
  const limit = 10
  const cookies = new Cookies();
  const [page, setPage] = useState(1)
  const [images, loading, hasError, readData] = useFetch(`${endPoint}?page=${encodeURIComponent(page)}&limit=${encodeURIComponent(limit)}`)
  const [mode, setMode] = useState('gallery')
  const [selected, setSelected] = useState(null)
  const [newName, setNewName] = useState('')
  const [modalOpen, setModalOpen] = useState(false)

  const handleModeChange = (type) => {
    setMode(type)
  }

  const handleModal = (isOpen) => {
    setModalOpen(isOpen)
  }

  const handleSelectImage = (id) => {
    const image = images.find(image => image.id === id)
    setSelected(image)
    setNewName(image.name)
  }
  
  const handleEdit = (id) => {
    handleSelectImage(id)
    setModalOpen(true)
  }

  const handleChangeName = (value) => {
    setNewName(value)
  }

  const handleSaveName = () => {
    cookies.set('named-' + selected.id, newName, { path: '/' });
    setModalOpen(false)
    readData()
  }

  return (
    <div className="gallery-container">
      <div className="gallery-header">
        <h2 className="gallery-title">Regallery</h2>
        <div className="gallery-control">
          <PageControl page={page} max={3} onChange={setPage} className=""/>
          <ModeControl mode={mode} onChange={handleModeChange} />
        </div>
      </div>

      <div className="gallery-content">
        {hasError ? <Alert message="There was some errors on the endpoint, please be patient." className="mb-4"/> : ''}

        {(mode === 'gallery') ? <Gallery selected={selected} onClick={handleEdit} isLoading={loading} images={images}/> : <List selected={selected} onClick={handleEdit} isLoading={loading} images={images}/>}
      </div>

      <Modal isOpen={modalOpen}>
        <Input type="text" className="" placeholder="Rename Image" value={ newName } onChange={(val) => handleChangeName(val) }/>
        <ModalControl onSave={handleSaveName} onCancel={() => handleModal(false)}/>
      </Modal>
    </div>
  )
}

export default App;
