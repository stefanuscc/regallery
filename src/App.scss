@import "/src/_vars";

a {
  color: #333;
  text-decoration: none;
}

a:hover {
  color: #666;
  text-decoration: none;
}

.gallery-container {
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
  justify-content: space-around;
  margin: 0 auto;
  padding: 2em;  

  .gallery-header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 1.5rem;

    @media screen and (max-width: $small) {
      flex-direction: column;
    }

    @media screen and (min-width: $medium) {
      flex-direction: row;
    }

    .gallery-title {
      margin: 0;
      font-size: 1.8rem;

      @media screen and (max-width: $small) {
        margin-bottom: 10px;
      }
    }
  }

  .gallery-control {
    display: flex;
  }
}

.sr-only {
  display: none;
}

.icon {
  width: 1rem;
  height: 1rem;
}