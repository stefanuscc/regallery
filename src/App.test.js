import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';

afterEach(cleanup);

test('App screen to match snapshot', () => {
    const component = renderer.create(
        <App />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
