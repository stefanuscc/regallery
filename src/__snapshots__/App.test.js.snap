// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App screen to match snapshot 1`] = `
<div
  className="gallery-container"
>
  <div
    className="gallery-header"
  >
    <h2
      className="gallery-title"
    >
      Regallery
    </h2>
    <div
      className="gallery-control"
    >
      <span
        className="group "
      >
        <button
          className="button icon  "
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Previous
          </span>
          <svg
            className="icon"
            fill="none"
            stroke="currentColor"
            strokeWidth="2"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15 19l-7-7 7-7"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
          </svg>
        </button>
        <button
          className="button  active "
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Page 
            1
          </span>
          1
        </button>
        <button
          className="button   "
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Page 
            2
          </span>
          2
        </button>
        <button
          className="button   "
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Page 
            3
          </span>
          3
        </button>
        <button
          className="button icon  "
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Next
          </span>
          <svg
            className="icon"
            fill="none"
            stroke="currentColor"
            strokeWidth="2"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 5l7 7-7 7"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
          </svg>
        </button>
      </span>
      <span
        className="group "
      >
        <button
          className="button icon active "
          data-mode="gallery"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="icon"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5 3a2 2 0 00-2 2v2a2 2 0 002 2h2a2 2 0 002-2V5a2 2 0 00-2-2H5zM5 11a2 2 0 00-2 2v2a2 2 0 002 2h2a2 2 0 002-2v-2a2 2 0 00-2-2H5zM11 5a2 2 0 012-2h2a2 2 0 012 2v2a2 2 0 01-2 2h-2a2 2 0 01-2-2V5zM11 13a2 2 0 012-2h2a2 2 0 012 2v2a2 2 0 01-2 2h-2a2 2 0 01-2-2v-2z"
            />
          </svg>
        </button>
        <button
          className="button icon  "
          data-mode="list"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="icon"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M3 4a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zm0 4a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zm0 4a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zm0 4a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
              fillRule="evenodd"
            />
          </svg>
        </button>
      </span>
    </div>
  </div>
  <div
    className="gallery-content"
  >
    
    <div
      className="grid"
    >
      
      <div
        className="skeleton"
      >
        <div
          className="image"
        />
        <div
          className="title"
        />
        <div
          className="name"
        />
      </div>
      <div
        className="skeleton"
      >
        <div
          className="image"
        />
        <div
          className="title"
        />
        <div
          className="name"
        />
      </div>
      <div
        className="skeleton"
      >
        <div
          className="image"
        />
        <div
          className="title"
        />
        <div
          className="name"
        />
      </div>
      <div
        className="skeleton"
      >
        <div
          className="image"
        />
        <div
          className="title"
        />
        <div
          className="name"
        />
      </div>
      <div
        className="skeleton"
      >
        <div
          className="image"
        />
        <div
          className="title"
        />
        <div
          className="name"
        />
      </div>
      <div
        className="skeleton"
      >
        <div
          className="image"
        />
        <div
          className="title"
        />
        <div
          className="name"
        />
      </div>
      <div
        className="skeleton"
      >
        <div
          className="image"
        />
        <div
          className="title"
        />
        <div
          className="name"
        />
      </div>
      <div
        className="skeleton"
      >
        <div
          className="image"
        />
        <div
          className="title"
        />
        <div
          className="name"
        />
      </div>
      <div
        className="skeleton"
      >
        <div
          className="image"
        />
        <div
          className="title"
        />
        <div
          className="name"
        />
      </div>
      <div
        className="skeleton"
      >
        <div
          className="image"
        />
        <div
          className="title"
        />
        <div
          className="name"
        />
      </div>
    </div>
  </div>
  <div
    className="modal hidden "
  >
    <div
      className="dialog"
    >
      
      <div
        className="header"
      >
        
      </div>
      <div
        className="content"
      >
        <input
          className="field "
          onChange={[Function]}
          placeholder="Rename Image"
          type="text"
          value=""
        />
        <span
          className="group "
        >
          <button
            className="button  active "
            onClick={[Function]}
            type="button"
          >
            SAVE
          </button>
          <button
            className="button   "
            onClick={[Function]}
            type="button"
          >
            CANCEL
          </button>
        </span>
      </div>
    </div>
  </div>
</div>
`;
